src/aktualizr_primary/secondary.cc

```diff
@@ -0,0 +1,51 @@
+#include <unordered_map>
+
+#include "secondary.h"
+#include "secondary_config.h"
+
+#include "ipuptanesecondary.h"
+
+namespace Primary {
+
+using SecondaryFactoryRegistry =
+    std::unordered_map<std::string, std::function<std::shared_ptr<Uptane::SecondaryInterface>(const SecondaryConfig&)>>;
+
+static SecondaryFactoryRegistry sec_factory_registry = {
+    {IPSecondaryConfig::Type,
+     [](const SecondaryConfig& config) {
+       auto ip_sec_cgf = dynamic_cast<const IPSecondaryConfig&>(config);
+       return Uptane::IpUptaneSecondary::create(ip_sec_cgf.ip, ip_sec_cgf.port);
+     }},
+    //  {
+    //     Add another secondary factory here
+    //  }
+};
+
+static std::shared_ptr<Uptane::SecondaryInterface> createSecondary(const SecondaryConfig& config) {
+  return (sec_factory_registry.at(config.type()))(config);
+}
+
+void initSecondaries(Aktualizr& aktualizr, const boost::filesystem::path& config_file) {
+  if (!boost::filesystem::exists(config_file)) {
+    throw std::invalid_argument("Secondary ECUs config file does not exist: " + config_file.string());
+  }
+
+  auto secondary_configs = SecondaryConfigParser::parse_config_file(config_file);
+
+  for (auto& config : secondary_configs) {
+    try {
+      LOG_INFO << "Creating Secondary of type: " << config->type();
```

Comment:
"\n"?

Reply:
Actually LOG_XXX adds "\n" to the end of a message, so I am removing "\n" I added in other places.

Reply:
huh, good to know